### ISSUE TITLE
Discuss loss function inheritance

### ISSUE THREAD

**Issue by RasmusOrsoe** [graphnet-team/graphnet#244]
I think it's quite difficult to see how data flows through the loss functions when calling e.g. MSELoss(). One needs to peel back several quite abstract classes and one eventually arrives at https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module. It looks to me like the choice of having LossFunction inherit from _WeightedLoss only adds complexity, not functionality (please correct me if I'm wrong). It might be worth revisiting that choice - and if we stick with it, then perhaps we should try and make a detailed for-developers flowchart of the data flow.
